include/qemu/host-utils.h

```diff
@@ -631,6 +631,20 @@ static inline uint64_t usub64_borrow(uint64_t x, uint64_t y, bool *pborrow)
 #endif
 }
 
+/**
+ * Compare 2 128-bit unsigned integers, passed in as unsigned 64-bit pairs
+ *
+ * Returns:
+ * >0 if ahi|alo > bhi|blo,
+ * 0 if ahi|alo == bhi|blo,
+ * <0 if ahi|alo < bhi|blo
+ */
+static inline int ucmp128(uint64_t alo, uint64_t ahi, uint64_t blo, uint64_t bhi)
+{
+    return (ahi == bhi) ? (alo > blo ? 1 : (alo == blo ? 0 : - 1)) :
```

Comment:
again, minor nit, but since you have 3 ternary operators, I'd leave the results of the first in a different line, and ident it:


```suggestion
    return (ahi == bhi) ?
                  (alo > blo ? 1 : (alo == blo ? 0 : - 1)) :
                  (ahi > bhi ? 1 : -1);
```